core/state/pruner/pruner.go

```diff
@@ -233,6 +257,111 @@ func prune(snaptree *snapshot.Tree, root common.Hash, maindb ethdb.Database, sta
 	return nil
 }
 
+// Prune block body data
+func (p *BlockPruner) BlockPruneBackUp(name string, cache, handles int, backFreezer, namespace string, readonly bool) error {
+	//Back-up the necessary data within original ancient directory, create new freezer backup directory backFreezer
+	//db, err = rawdb.NewLevelDBDatabaseWithFreezer(root, cache, handles, backFreezer, namespace, readonly)
+	start := time.Now()
+	chainDb := p.db
+	chainDbBack, err := p.n.OpenDatabaseWithFreezer(name, cache, handles, backFreezer, namespace, readonly)
+	if err != nil {
+		log.Error("Failed to open ancient database: %v", err)
+		return err
+	}
+
+	//write back-up data to new chainDb
+	// Restore the last known head block
+
+	//write genesis block firstly
+	genesis := p.genesis
+	if _, _, err := core.SetupGenesisBlock(chainDbBack, genesis); err != nil {
+		log.Error("Failed to write genesis block: %v", err)
+		return err
```

Comment:
The code in this commit was already outdated and invalid. There's no manually db closing anymore for updated code, it will close automatically by `defer db.Close()`